_includes/api/en/5x/routing-args.html

```diff
@@ -26,18 +26,19 @@ <h4> Arguments</h4>
 <tr>
 <td> <code>callback</code></td>
 <td>
-Callback functions; can be:
+A series of one or more functions that are invoked with the following arguments: <code>request</code>, <code>response</code>, <code>next</code>. Each callback optionally sends a response to the client or passes control to the next callback in the series by calling `next()`. `callback` can be provided as:
 <ul>
-<li>A middleware function.</li>
-<li>A series of middleware functions (separated by commas).</li>
+<li>One or more middleware functions.</li>
 <li>An array of middleware functions.</li>
-<li>A combination of all of the above.</li>
+<li>Any combination of the above.</li>
 </ul>
 <p>
-You can provide multiple callback functions that behave just like middleware, except
-that these callbacks can invoke <code>next('route')</code> to bypass
-the remaining route callback(s). You can use this mechanism to impose pre-conditions
-on a route, then pass control to subsequent routes if there is no reason to proceed with the current route.
+The callbacks are invoked in the order in which they are provided and behave in the same way as middleware. In order for the next callback to be invoked, the current one must call <code>next()</code>. Every callback must either send a response, or call next(), allowing one of the subsequent callbacks to send a response. If execution of the series of callbacks ends without sending a response to the client (or erroring), the request will hang.
+</p>
+<p>
+Throwing an error or returning a rejected promise in a middleware function will cause all further route callbacks to be skipped. This allows a middleware function to be defined using the <code>async</code> keyword and to <code>await</code> on promises.</p>
```

Comment:
I think I'd prefer to call these "async functions" instead of "defined using the async keyword" so like:

> This allows middleware to be defined using async functions

 And then maybe include as part of it some kind of explanation like this: 

> Whenever `await` is called on a promise that is later rejected, the middleware chain will be broken and the default error handler will be called with the rejected value, similar to calling `return next(err)` from inside a middleware function. 

I don't know if it needs to be called out explicitly but because of this architecture it's not necessary to wrap every express middleware or route `await` in `try/catch` which is common in most examples.

Reply:
Okay that last part is covered pretty well below, sorry!

Reply:
@xjamundx thanks for reviewing this. I've pushed an update with b369599 that attempts to simplify the wording, please let me know what you think.

Reply:
Yeah that's good. I think there's a meta discussion about what's a "route" vs a "route callback" vs a "middleware function", but I don't think we need to figure that out right now. This is good

Reply:
I completely agree. It would be nice if somewhere (maybe the glossary) explains the differences between route callback and middleware. Do you know what they are? The only documented difference I'm aware of is that a route callback can call `next('route')` to skip the remaining route callbacks.

Reply:
I think it's simply detected in the code base based on the number of arguments your handler includes. If it's 3 it's a middleware, 4 it's an error handler. Definitely something to call out clearly somewhere.